### PR TITLE
Carrier traps and acid wells are destroyed when weeds below them is removed

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

fixes https://github.com/tgstation/TerraGov-Marine-Corps/issues/6985

Acid wells are destroyed when weeds below them is removed

Move all xeno structures to xeno_structure.dm 

## Changelog
:cl:
fix: Fix carrier trap not being killed when weeds are destroyed
balance: Acid wells are destroyed when the weeds below them is removed
/:cl:

### CODE REVIEW

**Comment by InterroLouis** [comment]
What about acid wells?

**Comment by BraveMole** [comment]
> What about acid wells?

ah good point, forgot about those

**Comment by Ieteihowmee** [comment]
Finally! I always thought it was weird most of them did not just break.

**Comment by Surrealaser** [comment]
Man, this is going to make it way too easy to clear out xeno structures.

Also tunnels should not be subject to this if they are.

**Comment by QualityVan** [comment]
Tunnels don't require resin to place one initially, it doesn't make sense that they'd die without it.

**Comment by Ragecrabbo** [comment]
This is going to be a pretty big nerf to xeno structures but a necessary one (atleast for the carrier traps)
Hopefully something will be done to the costs to make up for it as i can see a lot of situations where a marine destroys one weed node and takes out multiple structures alongside it. Traps especially which rely on marines stepping them will have to have multiple nodes around them if you want to ensure they stay around long enough for someone to trigger them.

**Comment by SplinterGP** [comment]
> Also tunnels should not be subject to this if they are (and they seem to be); it's a literal hole that has no reliance on resin whatsoever.
> 
> Jelly pods should probably not be subject to this given the cooldown involved with making them and their relative durability. If all you need to do to get rid of the things is just destroy the resin beneath them, that's pretty fucked.

^

**Review by TiviPlus — CHANGES_REQUESTED** [review]
make tunnels and jelly pods immune

**Comment by BraveMole** [comment]
Tests are succeeding btw

**Comment by turnop** [comment]
good pr

carrier traps that embed themselves into bedsheets are bull

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -132,3 +132,6 @@ GLOBAL_VAR_INIT(global_unique_id, 1)
 #define LIGHT_EMPTY 1
 #define LIGHT_BROKEN 2
 #define LIGHT_BURNED 3
+
+//Xeno structure flags
+#define IGNORE_WEED_REMOVAL (1<<0)
```
Comment (TiviPlus) [inline code/__DEFINES/misc.dm:137]: This should not be in misc it shuld use the proper file

File: code/modules/xenomorph/xeno_structures.dm [hunk 2]
```diff
@@ -2,6 +2,570 @@
...
+	X.put_in_active_hand(hugger)
+	hugger.go_active(TRUE)
+	hugger = null
+	to_chat(X, "<span class='xenonotice'>We remove the facehugger from [src].</span>")
+
+/obj/structure/xeno/trap/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(istype(I, /obj/item/clothing/mask/facehugger) && isxeno(user))
+		var/obj/item/clothing/mask/facehugger/FH = I
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:122]: early return

File: code/modules/xenomorph/xeno_structures.dm [hunk 3]
```diff
@@ -2,6 +2,570 @@
...
+	if(!C.can_be_facehugged(hugger))
+		return
+	playsound(src, "alien_resin_break", 25)
+	C.visible_message("<span class='warning'>[C] trips on [src]!</span>",\
+						"<span class='danger'>You trip on [src]!</span>")
+	C.Paralyze(4 SECONDS)
+	xeno_message("A facehugger trap at [AREACOORD_NO_Z(src)] has been triggered!", "xenoannounce", 5, hugger.hivenumber,  FALSE, get_turf(src), 'sound/voice/alien_talk2.ogg', FALSE, null, /obj/screen/arrow/attack_order_arrow, COLOR_ORANGE, TRUE) //Follow the trend of hive wide alerts for important events
+	drop_hugger()
+
+/obj/structure/xeno/trap/proc/drop_hugger()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:94]: autodoc

File: code/modules/xenomorph/xeno_structures.dm [hunk 4]
```diff
@@ -2,6 +2,570 @@
...
+		icon_state = "trap0"
+	..()
+
+/obj/structure/xeno/trap/fire_act()
+	if(hugger)
+		hugger.forceMove(loc)
+		hugger.kill_hugger()
+		hugger = null
+		icon_state = "trap0"
+	..()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:77]: why are return values not preserved here

File: code/modules/xenomorph/xeno_structures.dm [hunk 5]
```diff
@@ -2,6 +2,570 @@
...
+
+/obj/structure/xeno/trap/examine(mob/user)
+	. = ..()
+	if(isxeno(user))
+		to_chat(user, "A hole for a little one to hide in ambush.")
+		if(hugger)
+			to_chat(user, "There's a little one inside.")
+		else
+			to_chat(user, "It's empty.")
+
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:62]: early returns

File: code/modules/xenomorph/xeno_structures.dm [hunk 6]
```diff
@@ -2,6 +2,570 @@
...
+	resistance_flags = UNACIDABLE
+	layer = RESIN_STRUCTURE_LAYER
+
+	max_integrity = 140
+
+	hud_possible = list(XENO_TACTICAL_HUD)
+	xeno_structure_flags = IGNORE_WEED_REMOVAL
+
+	var/tunnel_desc = "" //description added by the hivelord.
+	var/mob/living/carbon/xenomorph/hivelord/creator = null
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:158]: Autodoc

File: code/modules/xenomorph/xeno_structures.dm [hunk 7]
```diff
@@ -2,6 +2,570 @@
...
+	. = ..()
+	GLOB.xeno_tunnels += src
+	prepare_huds()
+	for(var/datum/atom_hud/xeno_tactical/xeno_tac_hud in GLOB.huds) //Add to the xeno tachud
+		xeno_tac_hud.add_to_hud(src)
+	hud_set_xeno_tunnel()
+	SSminimaps.add_marker(src, src.z, MINIMAP_FLAG_XENO, "xenotunnel")
+
+/obj/structure/xeno/tunnel/Destroy()
+	var/drop_loc = get_turf(src)
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:173]: cast this

File: code/modules/xenomorph/xeno_structures.dm [hunk 8]
```diff
@@ -2,6 +2,570 @@
...
+			if(double_check == "Pick another tunnel")
+				return targettunnel.pick_a_tunnel(M)
+			else //Whether we say yes or cancel out of it
+				M.forceMove(targettunnel.loc)
+				M.visible_message("<span class='xenonotice'>\The [M] pops out of \the [src].</span>", \
+				"<span class='xenonotice'>We pop out through the other side!</span>")
+		else
+			to_chat(M, "<span class='warning'>\The [src] ended unexpectedly, so we return back up.</span>")
+	else
+		to_chat(M, "<span class='warning'>Our crawling was interrupted!</span>")
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:305]: early return

File: code/modules/xenomorph/xeno_structures.dm [hunk 9]
```diff
@@ -2,6 +2,570 @@
...
+	anchored = TRUE
+	max_integrity = 5
+	layer = RESIN_STRUCTURE_LAYER
+
+	hit_sound = "alien_resin_move"
+	destroy_sound = "alien_resin_move"
+
+	var/charges = 1
+	var/charging = FALSE
+	var/mob/living/carbon/xenomorph/creator = null
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:333]: autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/xenomorph/xeno_structures.dm [hunk 10]
```diff
@@ -1,7 +1,580 @@
+//Xeno structure flags
+#define IGNORE_WEED_REMOVAL (1<<0)
+
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:3]: Put this in a defines file where it belongs

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest is fine

**Pull request merged**